sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -27,11 +21,79 @@ AZ_NODISCARD az_result az_keyvault_keys_createKey(
     az_span keyname,
     az_span keytype,
     az_keyvault_keys_keys_options * options,
-    az_span * out) {
+    const az_mut_span * const out) {
   (void)client;
   (void)keyname;
   (void)keytype;
   (void)options;
   (void)out;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_keyvault_build_url(
+    az_span const uri,
+    az_span const key_type,
+    az_span const key_name,
+    az_mut_span const out) {
+  az_span_builder s_builder = az_span_builder_create(out);
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, uri));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, AZ_KEY_VAULT_URL_PATH_SEPARATOR));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, key_type));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, AZ_KEY_VAULT_URL_PATH_SEPARATOR));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, key_name));
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_keyvault_keys_getKey(
+    az_keyvault_keys_client const * const client,
+    az_span const keyname,
+    az_key_vault_key_type const keytype,
+    const az_mut_span * const out) {
+  // create request buffer TODO: define size for a getKey Request
+  uint8_t request_buffer[1024 * 4];
+  az_mut_span request_buffer_span = AZ_SPAN_FROM_ARRAY(request_buffer);
+
+  /* ******** build url for request  ******
+   * add keytype, keyname and version to url request
+   */
+  az_span const az_key_type_span = az_keyvault_get_key_type_span(keytype);
+
+  // make sure we can handle url within the MAX_URL_SIZE
+  size_t const url_size_with_path = client->uri.size + (AZ_KEY_VAULT_URL_PATH_SEPARATOR.size * 2)
+      + az_key_type_span.size + keyname.size;
+  AZ_CONTRACT(url_size_with_path < MAX_URL_SIZE, AZ_ERROR_BUFFER_OVERFLOW);
+
+  // stack a new buffer to hold url with key name and type
```

Comment:
Typo: a--> allocate?

Reply:
OK but the grammar isn't quite right; how about: 
Put a new buffer on the stack to hold a url with the key name and type

Reply:
Done